sick_lidar3d:
  ros__parameters:
    udp_sender : ""                    # Use "" (default) to receive msgpacks from any udp sender, use "127.0.0.1" to restrict to localhost (loopback device), or use the ip-address of a MRS100 lidar or MRS100 emulator
    udp_port : 2115                    # default udp port for multiScan136 resp. multiScan136 emulator is 2115
    publish_topic : "/cloud"           # ros topic to publish received msgpack data converted top PointCloud2 messages, default: "/cloud"
    publish_topic_all_segments: "/cloud_360" # ros topic to publish PointCloud2 messages of all segments (360 deg), default: "/cloud_360"
    segment_count: 12                  # number of expected segments in 360 degree, multiScan136: 12 segments, 30 degree per segment
    publish_frame_id : "world"         # frame id of ros PointCloud2 messages, default: "world"
    udp_input_fifolength : 20          # max. udp input fifo length(-1: unlimited, default: 20 for buffering 1 second at 20 Hz), elements will be removed from front if number of elements exceeds the fifo_length
    msgpack_output_fifolength : 20     # max. msgpack output fifo length(-1: unlimited, default: 20 for buffering 1 second at 20 Hz), elements will be removed from front if number of elements exceeds the fifo_length
    verbose_level : 1                  # verbose_level <= 0: quiet mode, verbose_level == 1: print statistics, verbose_level == 2: print details incl. msgpack data, default: 1
    measure_timing : True              # measure_timing == true: duration and latency of msgpack conversion and export is measured, default: true
    visualize : False                  # visualize == true: plot lidarpoints via python wrapper, visualize == false: no diagrams, default: false
    export_csv : False                 # export msgpack data to csv file, default: false
    export_udp_msg : False             # true : export binary udpand msgpack data to file(*.udp and* .msg), default: false
    exit_on_keys_esc_q : True          # exit lidar3d_mrs100_recv after pressing key ESC, 'q' or 'Q'
    logfolder : "./logfiles"           # output folder for logfiles, default: "."
    inputfolder_msgfiles : "/tmp/mrs100dataparser-1.1.1R-msgpacks" # Input folder with msgpack files *.msg in offline mode (read from file, decode and optionally export to csv)
    input_msgfiles : ""                # Input msgpack files *.msg to read in offline/test mode
    mrs100_post_start_stop : False     # Post start and stop commands to start/stop multiScan136 using Rest-API
    mrs100_ip : "192.168.0.1"          # IP address of multiScan136
    mrs100_dst_ip : ""                 # UDP destination IP address (ip address of udp receiver)
    mrs100_port : 2115                 # IP port of multiScan136 to post start and stop commands using Rest-API
    mrs100_send_udp_start: False       # Send udp start string to multiScan136
    mrs100_send_udp_start_string: "magicalActivate" # udp string to start multiScan136
    # SOPAS settings:
    sopas_tcp_port : "2111"                 # TCP port for SOPAS commands, default port: 2111
    start_sopas_service : True              # True: sopas services for CoLa-commands are started (ROS only), default: true
    send_sopas_mrs100_start_stop_cmd : True # True: multiScan136 start and stop command sequence ("sWN ScanDataEnable 0/1" etc.) are sent after driver start and stop, default: true
    sopas_cola_binary : False               # False: SOPAS uses CoLa-A (ascii, default, recommended), CoLa-B (true, binary) currently experimental
    sopas_timeout_ms : 5000                 # Timeout for SOPAS response in milliseconds, default: 5000
    client_authorization_pw : "F4724744"    # Default password for client authorization
    # MSR100 filter settings
    mrs100_read_filtersettings: True        # Read multiScan136 settings for FREchoFilter, LFPangleRangeFilter and LFPlayerFilter at startup, default: true
    mrs100_FREchoFilter: 0                  # Optionally set FREchoFilter with 0 for FIRST_ECHO (default, EchoCount=1), 1 for ALL_ECHOS (EchoCount=3), or 2 for LAST_ECHO (EchoCount=1)
    mrs100_set_FREchoFilter: False          # If true, FREchoFilter is set at startup (default: false)
    mrs100_LFPangleRangeFilter: "0 -180.0 +179.0 -90.0 +90.0 1" # Optionally set LFPangleRangeFilter to "<enabled> <azimuth_start> <azimuth_stop> <elevation_start> <elevation_stop> <beam_increment>" with azimuth and elevation given in degree
    mrs100_set_LFPangleRangeFilter: False                       # If true, LFPangleRangeFilter is set at startup (default: false)
    mrs100_LFPlayerFilter: "0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1"  # Optionaly set LFPlayerFilter to "<enabled> <layer0-enabled> <layer1-enabled> <layer2-enabled> ... <layer15-enabled>" with 1 for enabled and 0 for disabled
    mrs100_set_LFPlayerFilter: False                            # If true, LFPlayerFilter is set at startup (default: false)
    # msgpack validation
    msgpack_validator_enabled : False # true: check msgpack data for out of bounds and missing scan data, false: no msgpack validation (default)
    msgpack_validator_verbose : 1     # 0: print error messages, 1: print error and informational messages, 2: print error and all messages
    msgpack_validator_discard_msgpacks_out_of_bounds : True # true: msgpacks are discarded if scan data out of bounds detected, false: error message if a msgpack is not validated
    msgpack_validator_check_missing_scandata_interval : 12 # 12 # check msgpack for missing scandata after collecting N msgpacks, default: N = 12 segments. Increase this value to tolerate udp packet drops. Use 12 to check each full scan.
    msgpack_validator_required_echos : "0"     # "0" # default: echo count = 1
    msgpack_validator_azimuth_start : -180.0   # default for full scan: -180 degree
    msgpack_validator_azimuth_end : +180.0     # default for full scan: +180 degree
    msgpack_validator_elevation_start : -90.0  # default for full scan: -90 degree
    msgpack_validator_elevation_end : +90.0    # default for full scan: +-90 degree
    msgpack_validator_valid_segments : "0 1 2 3 4 5 6 7 8 9 10 11"     # indices of valid segmentes, default for full scan: 12 segments
    msgpack_validator_layer_filter : "1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1" # default for full scan: 16 layer active, i.e. "1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1"

